engine/src/main/java/org/terasology/persistence/typeHandling/gson/GsonFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.persistence.typeHandling.gson;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapterFactory;
+import org.terasology.persistence.typeHandling.TypeSerializationLibrary;
+
+/**
+ * Class containing static factory methods for generating {@link Gson} objects that follow Terasology
+ * serialization rules and support Terasology TypeHandlers.
+ */
+public class GsonFactory {
+    /**
+     * Create a {@link GsonBuilder} with options set to comply with Terasology JSON serialization rules.
+     */
+    public static GsonBuilder createDefaultGsonBuilder() {
+        return new GsonBuilder()
+                .setExclusionStrategies(new GsonMapExclusionStrategy());
+    }
+
+    /**
+     * Create a {@link Gson} object which uses type handlers loaded from the given
+     * {@link TypeSerializationLibrary} and complies with Terasology JSON serialization rules.
+     *
+     * @param typeSerializationLibrary The {@link TypeSerializationLibrary} to load type handler
+     *                                 definitions from
+     */
+    public static Gson createGsonWithTypeSerializationLibrary(TypeSerializationLibrary typeSerializationLibrary) {
+        TypeAdapterFactory typeAdapterFactory =
+                new GsonTypeSerializationLibraryAdapterFactory(typeSerializationLibrary);
+
+        return createDefaultGsonBuilder()
+                .registerTypeAdapterFactory(typeAdapterFactory)
+                .create();
+    }
+
+    /**
+     * Create a {@link Gson} object which uses the given type handlers and complies with Terasology
+     * JSON serialization rules.
+     *
+     * @param typeHandlerEntries The type handlers to use during serialization.
+     */
+    @SuppressWarnings("unchecked")
+    public static Gson createGsonWithTypeHandlers(TypeHandlerEntry<?>... typeHandlerEntries) {
```

Comment:
`TypeHandlerEntry` is used to ensure that the `TypeHandler` actually is a type handler for the intended type, because this check is not performed manually further down the line. On creation of a `TypeHandlerEntry`, the compiler ensures that the `TypeHandler` passed is generic over the same type as is specified by the `class`.